proxy/coco/src/project/create.rs

```diff
@@ -18,6 +15,27 @@ use radicle_surf::vcs::git::git2;
 
 use crate::{config, user::User};
 
+/// Validation Errors
+pub mod validation {
+    use std::path::PathBuf;
+
+    /// Errors that occur when validating a [`super::Repo`]'s path.
+    #[derive(Debug, thiserror::Error)]
+    pub enum Error {
+        /// The path already existed when trying to create a new project.
+        #[error("the path provided '{0}' already exists")]
+        AlreadExists(PathBuf),
+
+        /// The path was expected to exist already but does not.
+        #[error("the path provided '{0}' does not exist when it was expected to")]
+        PathDoesNotExist(PathBuf),
+
+        /// The path was expected to point to a git repository but it did not.
```

Comment:
Shouldn't this be git-agnostic? :thinking:

Reply:
I dunno how I can make it git agnostic unless you have some suggestions